crates/matrix-sdk/tests/integration/client.rs

```diff
@@ -391,17 +391,31 @@ async fn join_room_by_id() {
     Mock::given(method("POST"))
         .and(path_regex(r"^/_matrix/client/r0/rooms/.*/join"))
         .and(header("authorization", "Bearer 1234"))
-        .respond_with(ResponseTemplate::new(200).set_body_json(&*test_json::ROOM_ID))
+        .respond_with(
+            ResponseTemplate::new(200)
+                .set_body_json(json!({"room_id":  *test_json::DEFAULT_SYNC_ROOM_ID})),
```

Comment:
```suggestion
                .set_body_json(json!({ "room_id": *test_json::DEFAULT_SYNC_ROOM_ID })),
```

Reply:
Also why the `*`, is that really needed?

Reply:
I assumed it was because `test_json::DEFAULT_SYNC_ROOM_ID` is of type `Lazy`. Will try it out.

Reply:
Oh, that makes sense.

Reply:
it was necessary, I left it in.